modules/OpenXRJoypad_module/src/OpenXRJoypadModule.cpp

```diff
@@ -0,0 +1,1384 @@
+    for (int i = 0; i < 10 && !okPosition; i++)
+    {
+        okPosition = m_encodersInterface->getEncoders(m_positionFeedbackInDegrees.data());
+
+        if (!okPosition)
+            yarp::os::Time::delay(0.1);
+    }
+    if (!okPosition)
+    {
+        yError() << "[RobotControlHelper::configure] Unable to read encoders (position).";
+        return false;
+    }
+
+    if (!switchToControlMode(m_controlMode))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to switch the control mode";
+        return false;
+    }
+    return true;
+}
+
+bool RobotControlHelper::switchToControlMode(const int& controlMode)
+{
+    // check if the control interface is ready
+    if (!m_controlModeInterface)
+    {
+        yErrorOnce() << "[RobotControlHelper::switchToControlMode] ControlMode I/F not ready.";
+        return false;
+    }
+
+    // set the control interface
+    std::vector<int> controlModes(m_actuatedDOFs, controlMode);
+    if (!m_controlModeInterface->setControlModes(controlModes.data()))
+    {
+        yError()
+            << "[RobotControlHelper::switchToControlMode] Error while setting the controlMode.";
+    }
+    return true;
+}
+
+bool RobotControlHelper::setDirectPositionReferences(const yarp::sig::Vector& desiredPosition)
+{
+    if (m_positionDirectInterface == nullptr)
+    {
+        yErrorOnce()
+            << "[RobotControlHelper::setDirectPositionReferences] PositionDirect I/F not ready.";
+        return false;
+    }
+
+    if (desiredPosition.size() != m_actuatedDOFs)
+    {
+        yError() << "[RobotControlHelper::setDirectPositionReferences] Dimension mismatch between "
+                    "desired position vector and the number of controlled joints.";
+        return false;
+    }
+
+    // convert radiant to degree
+    for (int i = 0; i < m_actuatedDOFs; i++)
+        m_desiredJointValue(i) = iDynTree::rad2deg(desiredPosition(i));
+
+    // set desired position
+    m_positionDirectInterface->setPositions(m_desiredJointValue.data());
+
+    return true;
+}
+
+bool RobotControlHelper::setVelocityReferences(const yarp::sig::Vector& desiredVelocity)
+{
+    if (m_velocityInterface == nullptr)
+    {
+        yErrorOnce() << "[RobotControlHelper::setVelocityReferences] Velocity I/F not ready.";
+        return false;
+    }
+
+    if (desiredVelocity.size() != m_actuatedDOFs)
+    {
+        yError() << "[RobotControlHelper::setVelocityReferences] Dimension mismatch between "
+                    "desired velocity vector and the number of controlled joints.";
+        return false;
+    }
+
+    // convert radiant/s  to degree/s
+    for (int i = 0; i < m_actuatedDOFs; i++)
+        m_desiredJointValue(i) = iDynTree::rad2deg(desiredVelocity(i));
+
+    // since the velocity interface use a minimum jerk trajectory a very high acceleration is set in
+    // order to use it as velocity "direct" interface
+    yarp::sig::Vector dummy(m_actuatedDOFs, std::numeric_limits<double>::max());
```

Comment:
We may store this as an attribute of the class

Reply:
Good point! Done in https://github.com/robotology/walking-teleoperation/pull/130/commits/49b30a3bc6cd20d520ba044f28a8fc1f2fe90001